File: app/src/main/java/org/coscup/ccip/network/ErrorUtil.java
package org.coscup.ccip.network;

import org.coscup.ccip.model.Error;

import java.io.IOException;
import java.lang.annotation.Annotation;

import okhttp3.ResponseBody;
import retrofit2.Converter;
import retrofit2.Response;

public class ErrorUtil {

    public static Error parseError(Response<?> response) {
        Converter<ResponseBody, Error> converter =
            CCIPClient.getRetrofit().responseBodyConverter(Error.class, new Annotation[0]);

        Error error;

        try {
            error = converter.convert(response.errorBody());
        } catch (IOException e) {
            return new Error();
        }

        return error;
    }
}

File: app/src/main/java/app/opass/ccip/network/ErrorUtil.kt
package app.opass.ccip.network

import app.opass.ccip.model.Error
import app.opass.ccip.util.JsonUtil
import retrofit2.Response
import java.io.IOException

class ErrorUtil {
    companion object {
        fun parseError(response: Response<*>): Error {
            return try {
                val body = response.errorBody()?.string() ?: return Error()
                JsonUtil.fromJson(body, Error::class.java)
            } catch (e: IOException) {
                Error()
            }
        }
    }
}
